Energy_Preserving.jl

```diff
@@ -0,0 +1,500 @@
+        end
+    end
+    contador = 0
+    #aqui almacenamos los 
+    auxiliar = Array{Int64}(undef, m)
+    longitud = length(array)
+    auxiliar[m] = longitud
+    #numero de subtrees
+    n = m-1
+    subtrees = Vector{Any}(undef, n)
+    for i in 2:longitud
+        if array[i] == 2
+            contador += 1
+            auxiliar[contador] = i
+            if contador == n
+                break
+            end
+        end
+    end
+    for i in 1:n
+        subtrees[i] = array[auxiliar[i]:(auxiliar[i+1]-1)]
+    end
+    push!(subtrees[n],array[longitud])
+    return subtrees
+end
+
+#println(BMinus(array))
+
+
+#this function returns a Hamiltonian H matriz of nxn
+
+
+              
+#funcion generadora de arboles simetricos
+function symmetries(tree)
+    subtrees = BMinus(tree)
+    #first order symmetries
+    perms = collect(permutations(subtrees))
+    l = length(perms)
+    for i in 1:l
+        perms[i] = reduce(vcat, perms[i])
+    end
+    result = add_one_to_left(perms)
+    return(result)
+end
+
+#generates all equivalent trees
+function equivalent_trees(array)
+    tree = rootedtree(array)
+    l = length(array)
+    superarray = get_permutations(array)
+    lperm = length(superarray)
+    for i in reverse(1:lperm) 
+        if superarray[i][1] != 1
+            splice!(superarray, i)
+        end
+    end
+    lperm = length(superarray)
+    flag = false
+    for i in reverse(1:lperm)
+        for j in 1:(l-1)
+            if (superarray[i][j+1] - superarray[i][j]) > 1
+                flag = true
+            end
+        end
+        if flag == true
+            splice!(superarray, i)
+        end
+        flag = false
+    end
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        arbol = rootedtree(superarray[i])
+        bandera = arbol == tree
+        if bandera == false
+            splice!(superarray, i)
+        end
+    end
+    bandera = false
+    lperm = length(superarray)
+    for i in reverse(1:lperm)
+        for j in 1:(i-1)
+            if superarray[i] == superarray[j]
+                bandera = true
+            end
+        end 
+        if bandera == true
+            splice!(superarray, i)
+        end
+        bandera = false
+    end
+
+    return superarray
+end
+
+function get_permutations(arr)
+    perms = collect(permutations(arr))
+    return perms
+end
+
+function num_leafs(a)
+    k = a[end]
+    return k - 1
+end
+
+function add_one_to_left(arrays)
+    for i in 1:length(arrays)
+        arrays[i] = vcat([1], arrays[i])
+    end
+    return arrays
+end
+
+#this function multplies the coefficient for its symmetry factor
+function symfact_normalization(coef,thetrees)
+    l = length(coef)
+    for i in 1:l
+        factor = 0
+        factor = RootedTrees.symmetry(RootedTree(thetrees[i]))
+        coef[i] = coef[i]*(1//factor)
+    end
+    return coef
+end
+
+#this function tells up to what order a method is Energy Preserving
+function OrderMethod(A,b)
```

Comment:
This looks strange to me
```julia
julia> # coefficients from your code with comment "stisfies for order <5, and not for 5"
       A = [0 0 0; 1//3 0 0; -5//48 15//16 0]; b = [1//10, 1//2, 2//5]; 

julia> EnergyPreserving(A, b, 1)
All the trees have been checked:
Condition Satisfied
Condition Satisfied

julia> EnergyPreserving(A, b, 2)
All the trees have been checked:
Condition Satisfied
Condition Satisfied

julia> EnergyPreserving(A, b, 3)
All the trees have been checked:
Condition Satisfied
Condition Satisfied

julia> EnergyPreserving(A, b, 4)
All the trees have been checked:
Condition Satisfied
Condition Satisfied

julia> EnergyPreserving(A, b, 5)
All the trees have been checked:
Condition Not Satisfied
Condition Not Satisfied

julia> OrderMethod(A, b)
All the trees have been checked:
Condition Satisfied
Energy Preserving for order < 1
```

Reply:
You are right: the 'if' cycle of the function 'OrderMethod' has to be fixed as I didn´t updated it with the rest of the functions:
if IsEnergyPreserving(trees,coefficients) == false
            flag = true
        end
The only difference is that in the first line I changed the 'true' for false. It should work with this modification.
I apologize for this.

Reply:
@Sondar74 You can make changes to the code on your computer, then commit and push them, and this PR will automatically be updated with the new code.